src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPriceEstimateTxInsertionManager.cs

```diff
@@ -0,0 +1,100 @@
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPriceEstimateTxInsertionManager : ITxInsertionManager
+    {
+        private readonly IGasPriceOracle _gasPriceOracle;
+        private readonly UInt256? _ignoreUnder;
+        private readonly UInt256 _baseFee;
+        private readonly ISpecProvider _specProvider;
+        public GasPriceEstimateTxInsertionManager(IGasPriceOracle gasPriceOracle, UInt256? ignoreUnder, 
+            UInt256 baseFee, ISpecProvider specProvider)
```

Comment:
Passing baseFee sugesst doing new object for every block... you can always do it with _blockTree.Head.BaseFee